### PR TITLE
WW-315 show todolist only for privileged users when not customized

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/WW-315

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/dc125c58cfa9e1a4f6ebcf77b30674686b51f812/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
2. MINOR [CommunityPageSpecialController.class.php#L42](https://github.com/Wikia/app/blob/dc125c58cfa9e1a4f6ebcf77b30674686b51f812/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L42): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
3. MINOR [CommunityPageSpecialController.class.php#L308](https://github.com/Wikia/app/blob/dc125c58cfa9e1a4f6ebcf77b30674686b51f812/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L308): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php [hunk 1]
```diff
@@ -258,6 +258,21 @@ public function getBenefitsModalData() {
 		] );
 	}
 
+	public function getCommunityTodoListData() {
+		$userCanEditinterface = $this->getUser()->isAllowed( 'editinterface' );
+		$data = ( new CommunityPageSpecialCommunityTodoListModel() )->getData();
+
+		return array_merge( $data, [
+			'showEditLink' => $userCanEditinterface,
+			'showTodoListModule' => $data[ 'haveContent' ] || $userCanEditinterface,
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php:267]: if this condition would be more complex I would move it to a separate function

**Comment by sebastianmarzjan** [comment]
LGTM

**Comment by idradm** [comment]
LGTM, however some refactoring would be nice here

**Pull request merged**